spec/models/alaveteli_pro/request_summary_spec.rb

```diff
@@ -28,4 +28,102 @@
     expect(summary.summarisable).to eq draft
     expect(draft.request_summary).to eq summary
   end
+
+  describe ".create_or_update_from" do
+    it "raises an ArgumentError if the request is of the wrong class" do
+      event = FactoryGirl.create(:info_request_event)
+      expect { AlaveteliPro::RequestSummary.create_or_update_from(event) }.
+        to raise_error(ArgumentError)
+    end
+
+    context "when the request already has a summary" do
+      it "updates the existing summary from a request" do
+        request = FactoryGirl.create(:info_request)
+        summary = FactoryGirl.create(:request_summary, summarisable: request)
+        body = FactoryGirl.create(:public_body)
+        request.title = "Updated title"
+        request.public_body = body
+        request.save
+        updated_summary = AlaveteliPro::RequestSummary.create_or_update_from(request)
+        expect(updated_summary.id).to eq summary.id
+        expect(updated_summary.title).to eq request.title
+        expect(updated_summary.public_body_names).to eq body.name
+        expect(updated_summary.summarisable).to eq request
+      end
+
+      it "updates the existing summary from a batch request" do
+        bodies = FactoryGirl.create_list(:public_body, 3)
```

Comment:
I've fixed this up (it was in a variety of places) in:
6092d93
87e200c
58af56b
1812be5
3e72a26
bcdfb4a
289a068
1c67ba9
a5744ff